File: diploma-server/src/test/java/org/arhor/diploma/service/MonsterReaderTest.java
package org.arhor.diploma.service;

import org.arhor.diploma.data.file.MonsterProvider;
import org.arhor.diploma.data.file.MonsterProviderImpl;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;
import org.springframework.core.io.ResourceLoader;
import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;

import static org.assertj.core.api.Assertions.assertThat;

@SpringJUnitConfig
class MonsterReaderTest {

    @TestConfiguration
    static class TestConfig {
        @Bean
        MonsterProvider monsterService(ResourceLoader loader) {
            return new MonsterProviderImpl(loader);
        }
    }

    @Autowired
    private MonsterProvider service;

    @Test
    void fetchAllMonsters() {
        // when
        var monsters = service.getList();

        // then
        assertThat(monsters)
                .isNotNull()
                .isNotEmpty();
    }
}

File: diploma-server/module-dnd/src/test/kotlin/org/arhor/diploma/dnd/data/repository/MonsterReaderTest.kt
package org.arhor.diploma.dnd.data.repository

import org.arhor.diploma.dnd.TestConfig
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Tag
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.junit.jupiter.SpringJUnitConfig

@Tag("integration-test")
@SpringJUnitConfig(TestConfig::class)
class MonsterReaderTest {

    @Autowired
    private lateinit var service: MonsterRepository

    @Test
    fun fetchAllMonsters() {
        // when
        val monsters = service.getPage()

        // then
        assertThat(monsters)
            .isNotNull

        assertThat(monsters.items)
            .isNotNull
            .isNotEmpty
    }
}
